client/scala/src/main/scala/ai/verta/client/Client.scala

```diff
@@ -47,11 +51,65 @@ class Client(conn: ClientConnection) {
 
   def getExperimentRun(id: String)(implicit ec: ExecutionContext) = {
     clientSet.experimentRunService.getExperimentRunById(Some(id))
-      .flatMap(r => if (r.experiment_run.isEmpty) Success(null) else Try[ExperimentRun]({
+      .flatMap(r => Try[ExperimentRun]({
         getExperiment(r.experiment_run.get.experiment_id.get) match {
           case Success(expt) => new ExperimentRun(clientSet, expt, r.experiment_run.get)
           case Failure(x) => throw x
         }
       }))
   }
+
+  /** Get the repository by name (and workspace). If not exist, create new repository
+   * @param name Name of the Repository
+   * @param workspace Workspace under which the Repository with name name exists. If not provided, the current user’s personal workspace will be used.
+   */
+  def getOrCreateRepository(name: String, workspace: Option[String] = None)(implicit ec: ExecutionContext) = {
+    GetOrCreateEntity.getOrCreate[Repository](
+      get = () => {
+        clientSet.versioningService.GetRepository(
+          id_named_id_workspace_name = if (workspace.isDefined) workspace.get else getPersonalWorkspace(),
+          id_named_id_name = urlEncode(name)
+        )
+        .map(r => new Repository(clientSet, r.repository.get))
+      },
+      create = () => {
+        clientSet.versioningService.CreateRepository(
+          id_named_id_workspace_name = if (workspace.isDefined) workspace.get else getPersonalWorkspace(),
+          body = VersioningRepository(
+            name = Some(name),
+            workspace_id = workspace
+          )
+        ).map(r => new Repository(clientSet, r.repository.get)
+        )
```

Comment:
Make sure to have consistent formatting. This formatting is different from the code 10 lines above it.

Reply:
resolved.

Reply:
resolved.